README.md

```diff
@@ -2,272 +2,396 @@
+As a matter of fact, let's do some of that together.
 
-#### Execution phase
+## Trying to Break the Scope Chain
 
-The second pass is the _execution phase_. The JavaScript engine again steps
-through our code line-by-line, but this time it actually runs our code,
-assigning values to variables and invoking functions.
+To further understand the scope chain, we must try to break it. That will let us
+see how much wiggle room the scope chain has, if any.
 
-One of the engine's tasks is the process of matching identifiers to the
-corresponding values stored in memory. Let's walk through the following code:
+What would happen if we try to invoke `addStrike()` _outside_ of `isOut()`?
 
 ```js
-const myVar = 42;
+const player = "Ohtani";
+
+function isOut(strikes) {
+  let atBat = true;
+  const availableStrikes = 3;
 
-myVar;
-// => 42
+  function addStrike() {
+    const totalStrikes = strikes + 1;
+    console.log(player + " has " + totalStrikes + " strike(s)!");
+
+    return totalStrikes;
+  }
+
+  if (strikes >= availableStrikes) {
+    atBat = false;
+    const message = player + " is out!";
+    console.log(message);
+  } else {
+    return addStrike();
+  }
+}
+
+addStrike(); // <= New addition
+isOut(1);
 ```
 
-During the compilation phase, a reference to the identifier `myVar` is stored in
-memory. The variable isn't yet assigned a value, and the second line (`myVar;`)
-is skipped over entirely because it isn't a declaration.
-
-During the execution phase, the value `42` is assigned to `myVar`. When the
-engine reaches the second line, it sees the identifier `myVar` and resolves it
-to a value through a process known as _identifier resolution_. The engine first
-checks the current scope to see if `myVar` has been declared in it. If it finds
-no declaration for `myVar` in the current scope, the engine then starts moving
-up the scope chain, checking the parent scope and then the parent scope's parent
-scope and so on until it finds a matching declared identifier or reaches the
-global scope. If the engine traverses all the way up to the global scope and
-still can't find a match, it will throw a `ReferenceError` and inform you that
-the identifier is not declared anywhere in the scope chain.
-
-Let's look at an example. Remember, the engine will continue to move up the
-scope chain **only** if it can't find a matching identifier in the current
-scope. Because of this, we can actually use the same identifier to declare
-variables or functions in multiple scopes:
+It throws a `ReferenceError`, telling us that `addStrike` does not exist. This
+is because the **scope chain can only go in one direction**. The chain can only
+be traversed starting from inwards and going outwards. It cannot be traveled
+from the outside in.
+
+In other words, **inner scopes _can_ access anything in its outer scopes. But
+outer scopes _cannot_ access anything in its inner scopes.**
+
+Let's try one more wacky thing. What if we try to access `totalStrikes` within
```

Comment:
I really don't think this lesson is too long. It all _seems_ relevant to me even though I'm no expert. If you do need to take something out though, you could consider taking out one of these walk through examples unless they are both necessary.